tests/valtio/atomWithProxy.test.tsx

```diff
@@ -130,3 +130,49 @@ it('state with a promise', async () => {
   fireEvent.click(getByText('button'))
   await findByText('status: modified')
 })
+
+it('synchronous atomWithProxy and regular atom ', async () => {
+  const proxyState: { elements: Record<string, string> } = proxy({
+    elements: {},
+  })
+
+  const stateAtom = atomWithProxy(proxyState, { sync: true })
+  const selectedElementIdAtom = atom('')
+
+  const createElementAtom = atom(null, (_, set) => {
+    const id = '123'
+    set(selectedElementIdAtom, id)
+    proxyState.elements[id] = `element`
+  })
+
+  const Elements = () => {
+    const [state] = useAtom(stateAtom)
+    const [selected] = useAtom(selectedElementIdAtom)
+    const [, create] = useAtom(createElementAtom)
+
+    return (
+      <>
+        <span>
+          selected element:{' '}
+          {selected === '' ? 'none' : state.elements[selected]}
+        </span>
+        <button
+          onClick={() => {
+            create()
+          }}>
+          create and select element
+        </button>
+      </>
+    )
+  }
+
+  const { findByText, getByText } = render(
+    <Provider>
+      <Elements />
+    </Provider>
+  )
+
+  await findByText('selected element: none')
+  fireEvent.click(getByText('create and select element'))
+  getByText('selected element: element') // synchronous
```

Comment:
I never tried this test, but it looks good. Let's do this for now, and revisit it in the future if something is necessary.

Reply:
the other way i had it working was with await but it was hard to read IMO bc i had to "invert" the test

https://github.com/pmndrs/jotai/pull/752/commits/ce31bd264dda39be6aa204ae86a39fabb29d8baf#diff-25a143e1f82ca88bb0daf47b0a2910c9974fbca024f2921fdbeebd321e205a2aR181-R183

Reply:
happy to whatver makes sense for u tho!